modules/catalog/src/main/java/org/apache/ignite/internal/catalog/storage/UpdateLogImpl.java

```diff
@@ -215,7 +218,7 @@ public CompletableFuture<Void> onUpdate(WatchEvent event) {
 
         @Override
         public void onError(Throwable e) {
-            assert false;
+            // Do nothing.
```

Comment:
When `WatchListener::onError` throws an error/exception, the operation never completes. Because `WatchProcessor::notifyWatches` does not catch exceptions thrown from `onError` method.